tool/teleport/common/teleport.go

```diff
@@ -1127,3 +1139,30 @@ func onIntegrationConfAccessGraphAWSSync(params config.IntegrationConfAccessGrap
 
 	return nil
 }
+
+func onIntegrationConfSAMLIdPGCPWorkforce(params samlidpconfig.GCPWorkforceAPIParams) error {
+	ctx := context.Background()
+
+	// Ensure we print output to the user. LogLevel at this point was set to Error.
+	utils.InitLogger(utils.LoggingForDaemon, slog.LevelInfo)
+
+	gcpWorkforceService, err := samlidp.NewGCPWorkforceService(samlidp.GCPWorkforceService{
+		APIParams: params,
+		HTTPClient: &http.Client{
```

Comment:
It seems we are creating a custom `CheckRedirect`. Is it really necessary?

Also, we have `defaults.HTTPClient()`. Can we use it here instead?

Reply:
I don't want the client to follow redirect here and expect metadata to be available at the given endpoint. 

added comment to make this clear, also used `defaults.HTTPClient()` https://github.com/gravitational/teleport/pull/39820/commits/c65c9083aa11a02d3b89f1cb9e82181149cb3596